sims/vcs/Makefile

```diff
@@ -55,18 +55,15 @@ VCS_CC_OPTS = \
 
 VCS_NONCC_OPTS = \
 	+lint=all,noVCDE,noONGS,noUI \
-	-error=PCWM-L \
```

Comment:
This is a port width mismatch error that I am removing for NVDLA. Now it just warn instead of error. If people want it kept in, I can have it be conditionally removed if NVDLA is present (like with `assert` for Ariane).

Reply:
Ya, in general I've found that I've had to make edits to the NVDLA RTL files to fix some illegal `wire` declarations and errors of that sort. Also, I seem to recall this width-mismatch-error, and that it came from files that aren't actually needed by the this particular configuration of the design. Perhaps is was in the bridge-DMA (BDMA) module? That's not part of either `nv_large` or `nv_small`. Something to keep in mind in general about the NVDLA RTL: not all of the RTL files that are generated are needed by the design, since your configuration might omit some features. And some of those files that aren't needed by the design sometimes contain illegal RTL (such as these width mismatches, and some undeclared signals). So maybe we want to `git rm` problematic files from the `vsrc` directory?

Anyway, what was the specific width-mismatch error you were seeing? It might be due to this situation.

Reply:
I just re-ran VCS with the error re-enabled and I found the error in the large config when the `NV_NVDLA_CSC_dl` was instantiated in the `NV_NVDLA_csc` module.

```
"NV_NVDLA_CSC_dl u_dl( .nvdla_core_clk (nvdla_op_gated_clk_2),  .nvdla_core_rstn (nvdla_core_rstn),
  The following 10-bit expression is connected to 1-bit port
  "sc2buf_dat_rd_shift" of module "NV_NVDLA_CSC_dl", instance "u_dl".
  Expression: sc2buf_dat_rd_shift
  The expression is from module NV_NVDLA_csc, instance u_NV_NVDLA_csc
```

Reply:
> Something to keep in mind in general about the NVDLA RTL: not all of the RTL files that are generated are needed by the design, since your configuration might omit some features. And some of those files that aren't needed by the design sometimes contain illegal RTL (such as these width mismatches, and some undeclared signals).

I also tried to prune all unnecessary files not needed to build the configs from the required files to reduce errors. So I believe the current set is the minimal workable set of requirements.

Reply:
Ah yes, I remember this error :)

This issue lies in `NV_NVDLA_CSC_dl.v`. Since the `CBUF_NO_SUPPORT_READ_JUMPING` macro is defined at the top of this file, the code on lines 3214 through 3216 is enabled. However, these three `wire` declarations with initializations are not legal, synthesizable verilog. This change should be made:

```diff
diff --git a/src/main/resources/vsrc/large/vmod/nvdla/csc/NV_NVDLA_CSC_dl.v b/src/main/resources/vsrc/large/vmod/nvdla/csc/NV_NVDLA_CSC_dl.v
index a2e2c1e..8f2cc21 100644
--- a/src/main/resources/vsrc/large/vmod/nvdla/csc/NV_NVDLA_CSC_dl.v
+++ b/src/main/resources/vsrc/large/vmod/nvdla/csc/NV_NVDLA_CSC_dl.v
@@ -3211,9 +3211,12 @@ assign dat_req_sub_h_1_addr_en = layer_st | ((dat_req_valid_d1 | dat_req_dummy_d
 assign dat_req_sub_h_2_addr_en = layer_st | ((dat_req_valid_d1 | dat_req_dummy_d1) & (dat_req_sub_h_d1 == 2'h2));
 assign dat_req_sub_h_3_addr_en = layer_st | ((dat_req_valid_d1 | dat_req_dummy_d1) & (dat_req_sub_h_d1 == 2'h3));
 `ifdef CBUF_NO_SUPPORT_READ_JUMPING
-wire sc2buf_dat_rd_next1_en = 1'b0;
-wire sc2buf_dat_rd_next1_en_w = 1'b0;
-wire sc2buf_dat_rd_shift = {10{1'b0}};
+wire sc2buf_dat_rd_next1_en;
+wire sc2buf_dat_rd_next1_en_w;
+wire [10 -1:0] sc2buf_dat_rd_shift;
+assign sc2buf_dat_rd_next1_en = 1'b0;
+assign sc2buf_dat_rd_next1_en_w = 1'b0;
+assign sc2buf_dat_rd_shift = {10{1'b0}};
 `endif
 `ifdef CBUF_SUPPORT_READ_JUMPING
 wire [10 -1:0] sc2buf_dat_rd_shift_w;
```

This same change is not necessary in the `nv_small` configuration, since while the same erroneous code exists, the relevant macro is not defined, so that code isn't enabled.